src/persist-types/src/error.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Codec related errors.
+
+use std::{error, fmt};
+
+/// A Codec related error.
+#[derive(Debug, Clone)]
+pub enum CodecError {
+    /// An unstructured codec related error.
+    String(String),
```

Comment:
Should this variant be called `Unstructured` like `CoordError::Unstructured`?

Reply:
I'm following the convention used in `persist::error::Error` which calls the unstructured variant `String`. No huge preference one way or the other